### PR TITLE
Remove bounding box query type parameter

### PR DESCRIPTION
**Pull request by iverase**
The parameter has been deprecates in 7.14 as it is a no-op.

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: docs/reference/migration/migrate_8_0/search.asciidoc [hunk 1]
```diff
@@ -173,5 +173,15 @@ If you query date fields without a specified `format`, check if the values in yo
...
+ .The `geo_bounding_box` query's `type` parameter has been removed.
+ [%collapsible]
+ ====
+ *Details* +
+ The `geo_bounding_box` query's `type` parameter was deprecated in 7.14.0 and has
+ been removed in 8.0.0. This parameter is a no-op and has no effect on the query.
+ *Impact* +
+ Discontinue use of the `type` parameter. `geo_bounding_box` queries that include
+ this parameter will return an error.
+ ====
```
Comment (jrodewig) [inline docs/reference/migration/migrate_8_0/search.asciidoc:187]: The indentation is causing this to render oddly.
Screen Shot 2021-06-24 at 9 03 25 AM

I'd remove the leading space to fix it.
```suggestion

.The `geo_bounding_box` query's `type` parameter has been removed.
[%collapsible]
====
*Details* +
The `geo_bounding_box` query's `type` parameter was deprecated in 7.14.0 and has
been removed in 8.0.0. This parameter is a no-op and has no effect on the query.

*Impact* +
Discontinue use of the `type` parameter. `geo_bounding_box` queries that include
this parameter will return an error.
====
```

**Review by jrodewig — COMMENTED** [review]
Looks good overall, but there's a formatting issue in the breaking changes we'll want to fix before merging. I left a suggestion to fix it.

Thanks @iverase!

**Review by jrodewig — APPROVED** [review]
Docs LGTM. Thanks @iverase.

**Review by imotov — APPROVED** [review]
LGTM.

**Pull request merged**